daemon/api_notices.go

```diff
@@ -213,3 +255,29 @@ func noticeViewableByUser(notice *state.Notice, requestUID uint32) bool {
 	}
 	return requestUID == userID
 }
+
+// noticeTypesViewableBySnap checks if passed interface allows the snap
+// to have read-access for the passed notice types.
+func noticeTypesViewableBySnap(types []state.NoticeType, r *http.Request) bool {
+	ucred, iface, err := ucrednetGetWithInterface(r.RemoteAddr)
+	if err != nil {
+		return false
+	}
+	if ucred.Socket != dirs.SnapSocket {
```

Comment:
Would it be better to change this to and explicit check of `ucred.Socket == dirs.SnapdSocket` instead since what if a 3rd snapd socket gets added - it would then be granted access here.

Reply:
Great catch, Thanks! Updated